Code/Tools/AssetProcessor/native/utilities/UuidManager.cpp

```diff
@@ -207,9 +232,48 @@ namespace AssetProcessor
         return newUuid;
     }
 
+    AZ::Outcome<void, AZStd::string> UuidManager::CacheUuidEntry(AZStd::string_view normalizedPath, AzToolsFramework::MetaUuidEntry entry, bool enabledType)
+    {
+        if (enabledType)
+        {
+            auto result = m_existingUuids.emplace(entry.m_uuid, normalizedPath);
+
+            if (!result.second)
+            {
+                // Insertion failure means this UUID is duplicated
+                return AZ::Failure(AZStd::string::format(
+                    "Source " AZ_STRING_FORMAT " has duplicate UUID " AZ_STRING_FORMAT " which is already assigned to another asset " AZ_STRING_FORMAT ". "
+                    "Every asset must have a unique ID.  Please change the UUID for one of these assets to resolve the conflict.",
+                    AZ_STRING_ARG(normalizedPath),
+                    AZ_STRING_ARG(entry.m_uuid.ToFixedString()),
+                    AZ_STRING_ARG(result.first->second)));
+            }
+        }
+
+        m_uuids[normalizedPath] = AZStd::move(entry);
+        return AZ::Success();
+    }
+
     AZ::Uuid UuidManager::CreateUuid()
     {
-        return AZ::Uuid::CreateRandom();
+        constexpr int MaxRetry = 50;
+
+        auto uuid = AZ::Uuid::CreateRandom();
+        int retry = 0;
+
+        while (m_existingUuids.contains(uuid) && retry < MaxRetry)
+        {
+            uuid = AZ::Uuid::CreateRandom();

```

Comment:
how often does it occur that you get a repeat?

Reply:
Most likely never, this is just in case